pkg/i2gw/providers/nginx/annotations/path_matching_test.go

```diff
@@ -0,0 +1,579 @@
+}
+
+func TestPathRegexCaseInsensitiveFlagInjection(t *testing.T) {
+	ingress := networkingv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-ingress",
+			Namespace: "default",
+			Annotations: map[string]string{
+				nginxPathRegexAnnotation: "case_insensitive",
+			},
+		},
+		Spec: networkingv1.IngressSpec{
+			IngressClassName: ptr.To("nginx"),
+			Rules: []networkingv1.IngressRule{
+				{
+					Host: "example.com",
+					IngressRuleValue: networkingv1.IngressRuleValue{
+						HTTP: &networkingv1.HTTPIngressRuleValue{
+							Paths: []networkingv1.HTTPIngressPath{
+								{
+									Path:     "/api/v[0-9]+",
+									PathType: ptr.To(networkingv1.PathTypePrefix),
+									Backend: networkingv1.IngressBackend{
+										Service: &networkingv1.IngressServiceBackend{
+											Name: "api-service",
+											Port: networkingv1.ServiceBackendPort{Number: 80},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Create HTTPRoute first (simulating what common converter creates)
+	routeName := common.RouteName(ingress.Name, ingress.Spec.Rules[0].Host)
+	routeKey := types.NamespacedName{Namespace: ingress.Namespace, Name: routeName}
+
+	originalPath := "/api/v[0-9]+"
+	ir := intermediate.IR{
+		HTTPRoutes: map[types.NamespacedName]intermediate.HTTPRouteContext{
+			routeKey: {
+				HTTPRoute: gatewayv1.HTTPRoute{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      routeName,
+						Namespace: ingress.Namespace,
+					},
+					Spec: gatewayv1.HTTPRouteSpec{
+						Rules: []gatewayv1.HTTPRouteRule{
+							{
+								Matches: []gatewayv1.HTTPRouteMatch{
+									{
+										Path: &gatewayv1.HTTPPathMatch{
+											Type:  ptr.To(gatewayv1.PathMatchPathPrefix),
+											Value: &originalPath,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Apply path regex feature
+	errs := PathRegexFeature([]networkingv1.Ingress{ingress}, nil, &ir)
+	if len(errs) > 0 {
+		t.Fatalf("Unexpected errors: %v", errs)
+	}
+
+	// Verify the path was modified
+	updatedRoute := ir.HTTPRoutes[routeKey]
+	if len(updatedRoute.HTTPRoute.Spec.Rules) == 0 {
+		t.Fatal("Expected HTTPRoute to have rules")
+	}
+
+	rule := updatedRoute.HTTPRoute.Spec.Rules[0]
+	if len(rule.Matches) == 0 {
+		t.Fatal("Expected HTTPRoute rule to have matches")
+	}
+
+	match := rule.Matches[0]
+	if match.Path == nil {
+		t.Fatal("Expected HTTPRoute match to have path")
+	}
+
+	// Verify path match type is regular expression
+	if match.Path.Type == nil || *match.Path.Type != gatewayv1.PathMatchRegularExpression {
+		t.Errorf("Expected PathMatchRegularExpression, got %v", match.Path.Type)
+	}
+
+	// Verify (?i) flag was injected
+	if match.Path.Value == nil {
+		t.Fatal("Expected path value to be set")
+	}
+
+	expectedPath := "(?i)/api/v[0-9]+"
+	if *match.Path.Value != expectedPath {
+		t.Errorf("Expected path value '%s', got '%s'", expectedPath, *match.Path.Value)
+	}
```

Comment:
nit: makes more sense this way, also same comment on t.fatal

```suggestion
	if match.Path.Value == nil {
		t.Fatal("Expected path value to be set")
	}

        // Verify (?i) flag was injected
	expectedPath := "(?i)/api/v[0-9]+"
	if *match.Path.Value != expectedPath {
		t.Errorf("Expected path value '%s', got '%s'", expectedPath, *match.Path.Value)
	}
```